packages/css/src/plugins/__tests__/atomicify-rules.test.tsx

```diff
@@ -0,0 +1,223 @@
+import postcss, { Plugin } from 'postcss';
+import whitespace from 'postcss-normalize-whitespace';
+import autoprefixer from 'autoprefixer';
+import { atomicifyRules } from '../atomicify-rules';
+
+const transform = (cssOrPlugins: TemplateStringsArray | Plugin<any>[]) => {
+  const result = postcss([atomicifyRules(), whitespace]).process(cssOrPlugins[0], {
+    from: undefined,
+  });
+
+  return result.css;
+};
+
+describe('atomicify rules', () => {
+  it('should atomicify a single declaration', () => {
+    const actual = transform`
+      color: blue;
+    `;
+
+    expect(actual).toMatchInlineSnapshot(`"._1doq13q2{color:blue}"`);
+  });
+
+  it('should should atomicify multiple declarations', () => {
+    const actual = transform`
+      color: blue;
+      font-size: 12px;
+    `;
+
+    expect(actual).toMatchInlineSnapshot(`"._1doq13q2{color:blue}._36l61fwx{font-size:12px}"`);
+  });
+
+  it('should autoprefix atomic rules', () => {
+    process.env.BROWSERSLIST = 'Edge 16';
+
+    const result = postcss([atomicifyRules(), whitespace, autoprefixer]).process(
+      'user-select: none;',
+      {
+        from: undefined,
+      }
+    );
+
+    expect(result.css).toMatchInlineSnapshot(`"._q4hxglyw{-ms-user-select:none;user-select:none}"`);
+  });
+
+  it('should callback with created class names', () => {
+    const classes: string[] = [];
+    const callback = (className: string) => {
+      classes.push(className);
+    };
+
+    const result = postcss([atomicifyRules({ callback }), whitespace, autoprefixer]).process(
+      'display:block;text-align:center;',
+      {
+        from: undefined,
+      }
+    );
+
+    result.css;
+
+    expect(classes).toMatchInlineSnapshot(`
+      Array [
+        "_dj7i1ule",
+        "_o3nk1h6o",
+      ]
+    `);
+  });
+
+  it('should atomicify a nested tag rule', () => {
+    const actual = transform`
+      div {
+        color: blue;
+      }
+    `;
+
+    expect(actual).toMatchInlineSnapshot(`"._k2hc13q2 div{color:blue}"`);
```

Comment:
🤔 Do we have to atomicify tags? Where we will add this class? Somewhere on top level?

Reply:
no we don't - the tag is included in the atomic group `_k2hc`

it gets assigned to the component, so its basically like:

```jsx
<div css={{ div: { color: 'blue' } }}>
  <div /> // color is blue
</div>
```

```jsx
<style>._k2hc13q2 div{color:blue}</style>
<div className="_k2hc13q2">
  <div /> // color is blue
</div>
```

Reply:
🤔

What about this usecase:

```typescript
<div css={{ div: { color: 'blue' }, span: { color: 'blue' } }}>
  <div /> <span />
</div>
```

If `color: blue` is used by both `div` and `span` tags? We will generate separate classnames for both?

Reply:
it will become this
`<style>._k2hc13q2 div,span{color:blue}</style>`

or `<style>._k2hc13q2 div{color:blue} ._k2hc13q2 span{color:blue}</style>` 🤔

Reply:
> Do we have to atomicify tags?
> the tag is included in the atomic group _k2hc
ahh we added it in the group. cool.

Reply:
yeah exactly 😄 should be okay. you can see an example of your use case in one of the tests too! :)

Reply:
ahh got ya..this is the one..cool. I didn't dig deep much. Will do later in the day 😄

```typescript
  it('should atomicify a nested multi selector rule', () => {
    const actual = transform`
      div, span, li {
        color: blue;
      }
    `;

    expect(actual).toMatchInlineSnapshot(
      `"._k2hc13q2 div, ._ijgx13q2 span, ._1jah13q2 li{color:blue}"`
    );
  });
```